Compiler crashes after issuing error for non-accessible annotation

```scala
import scala.annotation.migration
@migration("", "") class C
```

With 2.13, the compiler shows the error, then crashes. 2.12 is not affected.

```
➜  scalac Test.scala
Test.scala:2: error: class migration in package annotation cannot be accessed in package annotation
@migration("", "") class C
 ^
error: java.lang.NullPointerException
	at scala.tools.nsc.typechecker.Typers$Typer.$anonfun$typedAnnotation$1(Typers.scala:3940)
	at scala.tools.nsc.typechecker.Typers$Typer.typedAnnotation(Typers.scala:3834)
	at scala.tools.nsc.typechecker.Namers$Namer.$anonfun$annotSig$3(Namers.scala:1866)
	at scala.reflect.internal.AnnotationInfos$LazyAnnotationInfo.forcedInfo$lzycompute(AnnotationInfos.scala:158)
	at scala.reflect.internal.AnnotationInfos$LazyAnnotationInfo.forcedInfo(AnnotationInfos.scala:158)
	at scala.reflect.internal.AnnotationInfos$LazyAnnotationInfo.completeInfo(AnnotationInfos.scala:171)
	at scala.tools.nsc.typechecker.Typers$Typer.typedTemplate(Typers.scala:1998)
```

--- a/test/files/neg/t11607.check
+++ b/test/files/neg/t11607.check
@@ -1,0 +1,4 @@
+t11607.scala:2: error: class migration in package annotation cannot be accessed in package annotation
+@migration("", "") class C
+ ^
+one error found

--- a/test/files/neg/t11607.scala
+++ b/test/files/neg/t11607.scala
@@ -1,0 +1,2 @@
+import scala.annotation.migration
+@migration("", "") class C

